velox/functions/prestosql/MapRemoveNullValues.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+namespace facebook::velox::functions {
+
+template <typename TExec>
+struct MapRemoveNullValues {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  void call(
+      out_type<Map<Generic<T1>, Generic<T2>>>& out,
+      const arg_type<Map<Generic<T1>, Generic<T2>>>& inputMap) {
+    for (const auto& entry : inputMap) {
+      if (entry.second.has_value()) {
```

Comment:
In most cases there are no entries with null values or very few. It might be more efficient to avoid copying the map, but rather return it as is or wrap keys and values in a dictionary. But this requires writing this as a Vector function. Not suggesting to change anything, just sharing some thoughts.